tools/shared/src/main/scala/org/scalajs/core/tools/linker/EntryPoint.scala

```diff
@@ -0,0 +1,57 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.core.tools.linker
+
+import org.scalajs.core.ir.Definitions._
+import org.scalajs.core.ir.Types.ClassType
+
+import org.scalajs.core.tools.linker.analyzer.SymbolRequirement
+
+/** An entry point for a Scala.js application.
+ *
+ *  When linking a Scala.js application, a sequence of `EntryPoint`s can be
+ *  given. Those entry points will be executed at the startup of the
+ *  application. More specifically, the top-level code of the ECMAScript 2015
+ *  module emitted for the application will invoke the specified entry points
+ *  in the specified order, after having initialized everything else (notably
+ *  static initializers).
+ *
+ *  Instances of `EntryPoint` can be created with methods of
+ *  [[EntryPoint$ the EntryPoint companion object]].
+ */
+sealed abstract class EntryPoint
+
+/** Factory for [[EntryPoint]]s. */
+object EntryPoint {
```

Comment:
It is not ES terminology, no. In ES that's just code at the top-level of the module. It doesn't make sense from a Scala point of view.

I think "module initializer" is the best one. It's only overloaded internally. Static initializers are not available to user code, and objects being known as "modules" is also pretty internal.